mypy/stubgen.py

```diff
@@ -160,7 +160,8 @@ def find_module_path_and_all(module: str, pyversion: Tuple[int, int],
             module_all = getattr(mod, '__all__', None)
     else:
         # Find module by going through search path.
-        module_path = mypy.build.FindModuleCache().find_module(module, ['.'] + search_path,
+        module_path = mypy.build.FindModuleCache().find_module(module,
+                                                               ('.',) + tuple(search_path),
```

Comment:
@gvanrossum this is a bug on master. `lib_path` should be a tuple, and as you can see to the left, it is passed as a list. I can split this fix out if you don't want the rest.

Reply:
What is the concrete effect here? Could this cause problems to users?

Reply:
Essentially, `lru_cache` won't be able to hash the passed `list`.

```
$ python -m mypy.stubgen --no-import mypy
Traceback (most recent call last):
  File "C:\Python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\ethanhs\Documents\mypy\mypy\stubgen.py", line 1022, in <module>
    main()
  File "C:\Users\ethanhs\Documents\mypy\mypy\stubgen.py", line 901, in main
    raise e
  File "C:\Users\ethanhs\Documents\mypy\mypy\stubgen.py", line 898, in main
    include_private=options.include_private)
  File "C:\Users\ethanhs\Documents\mypy\mypy\stubgen.py", line 106, in generate_stub_for_module
    interpreter=interpreter)
  File "C:\Users\ethanhs\Documents\mypy\mypy\stubgen.py", line 164, in find_module_path_and_all
    interpreter)
TypeError: unhashable type: 'list'
```

Reply:
Then maybe make a separate PR for that change?

Reply:
OK, please split that out -- I can cherry-pick a 3-line fix to stubgen with a lot less risk and effort than this whole PR.

Reply:
@gvanrossum https://github.com/python/mypy/pull/4899